pkg/kubelet/pod_workers.go

```diff
@@ -766,10 +783,76 @@ func calculateEffectiveGracePeriod(status *podSyncStatus, pod *v1.Pod, options *
 	return gracePeriod, status.gracePeriod != 0 && status.gracePeriod != gracePeriod
 }
 
+// allowPodStart tries to start the pod and returns true if allowed, otherwise
+// it requeues the pod and returns false.
+func (p *podWorkers) allowPodStart(pod *v1.Pod) bool {
+	if !kubetypes.IsStaticPod(pod) {
+		// TBD: Do we want to allow non-static pods with the same full name?
+		// Note that it may disable the force deletion of pods.
+		return true
+	}
+	p.podLock.Lock()
+	defer p.podLock.Unlock()
+	status, ok := p.podSyncStatuses[pod.UID]
+	if !ok {
+		klog.ErrorS(nil, "Failed to get a valid podSyncStatuses", "pod", klog.KObj(pod), "podUID", pod.UID)
+		p.workQueue.Enqueue(pod.UID, wait.Jitter(p.backOffPeriod, workerBackOffPeriodJitterFactor))
+		status.working = false
+		return false
+	}
+	if !p.allowStaticPodStart(status.fullname, pod.UID) {
+		p.workQueue.Enqueue(pod.UID, wait.Jitter(p.backOffPeriod, workerBackOffPeriodJitterFactor))
+		status.working = false
+		return false
```

Comment:
If a static pod is never allowed to start, what happens? It appears that we keep requeueing it forever. Is that possible?

Reply:
Yes, it is possible if the previous static pod is terminating forever, but I think we are ensuring that the termination requested pod terminates after `terminationGracePeriodSeconds`.